src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -111,48 +116,20 @@ internal Conversions Conversions
         /// </summary>
         private HashSet<SyntaxTree> _lazyCompilationUnitCompletedTrees;
 
-        public override string Language
-        {
-            get
-            {
-                return LanguageNames.CSharp;
-            }
-        }
+        public override string Language => LanguageNames.CSharp;
 
-        public override bool IsCaseSensitive
-        {
-            get
-            {
-                return true;
-            }
-        }
+        public override bool IsCaseSensitive => true;
 
         /// <summary>
         /// The options the compilation was created with. 
         /// </summary>
-        public new CSharpCompilationOptions Options
-        {
-            get
-            {
-                return _options;
-            }
-        }
+        public new CSharpCompilationOptions Options => _options;
 
-        internal AnonymousTypeManager AnonymousTypeManager
-        {
-            get
-            {
-                return _anonymousTypeManager;
-            }
-        }
+        internal AnonymousTypeManager AnonymousTypeManager => _anonymousTypeManager;
 
-        internal override CommonAnonymousTypeManager CommonAnonymousTypeManager
-        {
-            get
-            {
-                return AnonymousTypeManager;
-            }
-        }
+        internal override CommonAnonymousTypeManager CommonAnonymousTypeManager => AnonymousTypeManager;
+
+        internal DelegateCacheManager DelegateCacheManager => _delegateCacheManager;
```

Comment:
Is it your intent to share the DelegateCacheManager across multiple Emit calls? If not, Compilation is probably not the right place to keep the instance. Since only module level containers can be potentially shared across Emit calls, I would reduce complexity and move this field/property to PEModuleBuilder.